calour/tests/test_manipulation.py

```diff
@@ -0,0 +1,50 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--,  Calour development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from unittest import main
+from copy import deepcopy
+
+import calour as ca
+
+from calour._testing import Tests, assert_experiment_equal
+
+
+class IOTests(Tests):
+    def setUp(self):
+        super().setUp()
+        self.test1 = ca.read(self.test1_biom, self.test1_samp)
+
+    def test_join_fields(self):
+        # test the default params
+        newexp = self.test1.join_fields('id', 'group', inplace=False)
+        self.assertIn('id_group', newexp.sample_metadata.columns)
+        self.assertEqual(newexp.sample_metadata.loc['S12', 'id_group'], '12.0_2')
+        # test we didn't change anything besides the new sample metadata column
+        assert_experiment_equal(newexp, self.test1, ignore_md_fields=['id_group'])
+
+        # test join feature fields with new field name, separator and inplace
+        exp = deepcopy(self.test1)
+        exp.join_fields('taxonomy', 'taxonomy', newname='test', axis=1, separator=';', inplace=True)
```

Comment:
split this func into 2 test func. could you also check the object with `self.assertIs` and `self.assertIsNot`